blottertrax/helper/repost_checker.py

```diff
@@ -0,0 +1,60 @@
+import sqlite3
+
+
+class RepostChecker:
+    cursor = None
+    sql = None
+
+    @classmethod
+    def __init__(self):
+        self.sql = sqlite3.connect('{}/../{}/../database/repost.db')
+        self.cursor = self.sql.cursor()
+    
+        self.cursor.execute('CREATE TABLE IF NOT EXISTS artistSubmission(artist TEXT UNIQUE NOT NULL, postTime REAL NOT NULL, repeatCount INT NOT NULL)')
+        self.cursor.execute('CREATE TABLE IF NOT EXISTS songSubmission(artistSong TEXT UNIQUE NOT NULL, postTime REAL NOT NULL)')
+        self.cursor.execute('CREATE TABLE IF NOT EXISTS idStamped(postID TEXT UNIQUE NOT NULL, postTime REAL NOT NULL)')
+        
+        self.cursor.execute('SELECT * FROM artistSubmission')
+    
+    @classmethod
+    def new_entry(self, artist_name, song_name, curr_time, postID):
+        self.cursor.execute('INSERT OR IGNORE INTO artistSubmission VALUES(?, ?, ?)', [artist_name, curr_time, 0])
+        self.cursor.execute('INSERT OR IGNORE INTO songSubmission VALUES(?, ?)', [song_name + '--||' + artist_name, curr_time])
+        self.cursor.execute('INSERT OR IGNORE INTO idStamped VALUES(?, ?)', [postID, curr_time])
+        self.sql.commit()
+    
+    @classmethod
+    def get_artist_timestamp(self, artist_name):
+        self.cursor.execute('SELECT postTime, repeatCount FROM artistSubmission WHERE artist == ?', [artist_name])
+        
+        return self.cursor.fetchone()
+    
+    @classmethod
+    def replace_entry(self, artist_name, song_name, curr_time, postID):
+        self.cursor.execute('UPDATE artistSubmission SET postTime=? WHERE artist == ?', [curr_time, artist_name])
+        self.cursor.execute('INSERT OR REPLACE INTO songSubmission VALUES(?, ?)', [song_name + '--||' + artist_name, curr_time])
+        self.cursor.execute('INSERT OR IGNORE INTO idStamped VALUES(?, ?)', [postID, curr_time])
+        self.sql.commit()
+    
+    @classmethod
+    def search_song(self, artist_name, song_name):
+        self.cursor.execute('SELECT postTime FROM songSubmission WHERE artistSong == ?', [song_name + '--||' + artist_name])
+    
+        return self.cursor.fetchone()
+    
+    @classmethod
+    def get_old_submissions(self, curr_time):

```

Comment:
> Think we should make this "get_submissions_older_than_time" or similar and have the math of
> 
> curr_time - 252000
> 
> done in the calling locations. This allows us to reuse this function if for some reason we want a different time period later.

as for this I agree it should be done, I'll do it in a bit.



> This would also required that we remove the
> 
> ```
>     self.cursor.execute('DELETE FROM idStamped WHERE postTime < ?', [curr_time - 252000])
>     self.sql.commit()
> ```
> 
> from this call and make it its own function so that we aren't accidently cleaning up the DB if we just want to query it for post info.

This one seems unnecessary to me, as we are always deleting whenever we get the data. The database is just a buffer to keep the data for 72 hours until we check for votes, the database itself serves no other purpose